splunk_otel/options.py

```diff
@@ -81,30 +80,32 @@ def __init__(
         access_token: Optional[str] = None,
         resource_attributes: Optional[Dict[str, Union[str, bool, int, float]]] = None,
         trace_response_header_enabled: Optional[bool] = None,
+        env: Optional[_EnvVarsABC] = _OSEnvVars(),
     ):
-        self._set_default_env()
-        self.access_token = self._get_access_token(access_token)
-        self.response_propagator = self._get_trace_response_header_enabled(
-            trace_response_header_enabled
+        self._set_default_env(env)
+        self.access_token = self._get_access_token(env, access_token)
+        self.response_propagator = self._get_response_propagator(
+            env, trace_response_header_enabled
         )
         self.resource = self._get_resource(service_name, resource_attributes)
         self.span_exporter_factories = self._get_span_exporter_factories(
-            span_exporter_factories
+            env, span_exporter_factories
         )
 
     @staticmethod
-    def _get_access_token(access_token: Optional[str]) -> Optional[str]:
+    def _get_access_token(env: _EnvVarsABC, access_token: Optional[str]) -> Optional[str]:
```

Comment:
> Why do you use _EnvVarsABC class here and not the _OSEnvVars?

This function signature just requires the `_EnvVarsABC` interface over the `_OSEnvVars` concrete implementation, but you're right, it can be changed to just accept the concrete implementation -- happy to change that.

> It starts with underscore, so it is supposed to be a private method, but it is nevertheless accessible. This change is breaking the default behavior for this function.

Since these are _private static methods inside a _private class, can you help me understand why you think this would be a breaking change? (I'm going to resolve the "same as above" comments below in this PR in reference to this concept so we can discuss and arrive at a decision in one place)